src/algorithms/configuration/maps/dense_map.py

```diff
@@ -17,68 +17,94 @@
+    def __init__(self, grid: Optional[List], services: Services = None) -> None:
+        self.grid = None 
 
-        if not grid:
+        arr_grid = None
+        if grid is not None:
+            arr_grid = np.atleast_2d(np.array(grid))
+            super().__init__(Size(*([0]*arr_grid.ndim)), services)
+        else:
+            super().__init__(Size(0, 0), services)
             return
 
-        self.set_grid(grid)
+        # Doesn't work with non-uniform grids
+        self.set_grid(arr_grid)
+
+    def set_grid(self, grid: np.array) -> None:
 
-    def set_grid(self, grid: List[List[int]]) -> None:
+        #We transpose here to not worry about flipping coordinates later on
+        #Please take care I'm not sure why everythng works but it does atm
+        np.transpose(grid)
         self.grid = grid
-        self.size = Size(len(grid[0]), len(grid))
-
-        for i in range(len(self.grid)):
-            for j in range(len(self.grid[i])):
-                if self.grid[i][j] == self.AGENT_ID:
-                    self.agent = Agent(Point(j, i))
-                if self.grid[i][j] == self.GOAL_ID:
-                    self.goal = Goal(Point(j, i))
-                if self.grid[i][j] == self.WALL_ID:
-                    self.obstacles.append(Obstacle(Point(j, i)))
-                if self.grid[i][j] == self.EXTENDED_WALL:
-                    self.obstacles.append(ExtendedWall(Point(j, i)))
+
+        self.size = Size(*self.grid.shape)
+        for index in np.ndindex(*self.size):
+            val: int = self.grid[index]
+            if val == self.AGENT_ID:
+                self.agent = Agent(Point(*index))
+            elif val == self.GOAL_ID:
+                self.goal = Goal(Point(*index))
+            elif val == self.WALL_ID:
+                self.obstacles.append(Obstacle(Point(*index)))
+            elif val == self.EXTENDED_WALL:
+                self.obstacles.append(ExtendedWall(Point(*index)))
 
     def extend_walls(self) -> None:
         """
         Method for extending the walls by agent radius
         """
 
+        #def extend_obstacle_bound() -> None:
+        #    for b in bounds:
+        #        for x in range(b.x - self.agent.radius, b.x + self.agent.radius + 1):
+        #            for y in range(b.y - self.agent.radius, b.y + self.agent.radius + 1):
+        #                if not self.is_out_of_bounds_pos(Point(x, y)):
+        #                    dist: Union[float, np.ndarray] = np.linalg.norm(np.array([x, y]) - np.array(b))
+        #                    if dist <= self.agent.radius and self.grid[y][x] == DenseMap.CLEAR_ID:
+        #                        self.grid[y][x] = DenseMap.EXTENDED_WALL
+        #                        self.obstacles.append(ExtendedWall(Point(x, y)))
+        #                        visited.add(Point(x, y))
+
         def extend_obstacle_bound() -> None:
             for b in bounds:
                 for x in range(b.x - self.agent.radius, b.x + self.agent.radius + 1):
```

Comment:
When this is merged, we need to raise an issue for this - extend_obstacle_bound is not 3d